public/components/custom_panels/home.tsx

```diff
@@ -0,0 +1,210 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+import { EuiGlobalToastList, EuiLink } from '@elastic/eui';
+import { Toast } from '@elastic/eui/src/components/toast/global_toast_list';
+import _ from 'lodash';
+import React, { ReactChild, useState } from 'react';
+import { CoreStart } from '../../../../../src/core/public';
+import {
+  CUSTOM_PANELS_API_PREFIX,
+  CUSTOM_PANELS_DOCUMENTATION_URL,
+} from '../../../common/constants/custom_panels';
+import { CustomPanelTable } from './custom_panels_table';
+
+// "Home" module is initial page for Custom Operantional Panels
+
+type Props = {
+  http: CoreStart['http'];
+  chrome: CoreStart['chrome'];
+  parentBreadcrumb: {text: string, href: string}[];
+};
+
+export type CustomPanelListType = {
+  name: string;
+  id: string;
+  dateCreated: string;
+  dateModified: string;
+};
+
+export const Home = ({ http, chrome, parentBreadcrumb }: Props) => {
+  const [customPanelData, setcustomPanelData] = useState<Array<CustomPanelListType>>([]);
+  const [openedCustomPanel, setopenedCustomPanel] = useState<CustomPanelListType | undefined>(
+    undefined
+  );
+  const [toasts, setToasts] = useState<Array<Toast>>([]);
+  const [loading, setLoading] = useState(false);
+
+  const setToast = (title: string, color = 'success', text?: ReactChild) => {
+    if (!text) text = '';
+    setToasts([...toasts, { id: new Date().toISOString(), title, text, color } as Toast]);
+  };
+
+  // Fetches all saved Custom Panels
+  const fetchCustomPanels = () => {
+    return http
+      .get(`${CUSTOM_PANELS_API_PREFIX}/`)
+      .then((res) => setcustomPanelData(res.panels))
+      .catch((err) => {
+        console.error('Issue in fetching the custom operational panels', err.body.message);
+      });
+  };
+
+  // Creates a new CustomPanel
+  const createCustomPanel = (newCustomPanelName: string) => {
+    if (newCustomPanelName.length >= 50 || newCustomPanelName.length === 0) {
```

Comment:
maybe move the validation logic into some utils module? so it's not defined multiple places and easier to change

Reply:
Oh yes this makes sense.  Will move this function to helpers.